src/core/mimetype/definitions.js

```diff
@@ -0,0 +1,544 @@
+                label: __('Flash video'),
+                equivalent: ['.flv'],
+                extensionsLabels: ['.flv']
+            },
+            {
+                mime: 'text/csv',
+                label: __('CSV file'),
+                equivalent: ['.csv'],
+                extensionsLabels: ['.csv']
+            },
+            {
+                mime: 'application/msword',
+                label: __('Microsoft Word'),
+                equivalent: [
+                    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+                    'application/vnd.ms-word.document.macroEnabled.12',
+                    'application/vnd.ms-word.template.macroEnabled.12',
+                    'application/vnd.openxmlformats-officedocument.wordprocessingml.template',
+                    '.doc',
+                    '.docx',
+                    '.dot',
+                    '.docm',
+                    '.dotm',
+                    '.dotx'
+                ],
+                extensionsLabels: ['.doc', '.docx']
+            },
+            {
+                mime: 'application/vnd.ms-excel',
+                label: __('Microsoft Excel'),
+                equivalent: [
+                    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+                    'application/vnd.ms-excel.sheet.binary.macroEnabled.12',
+                    'application/vnd.ms-excel.sheet.macroEnabled.12',
+                    '.xlsb',
+                    '.xlsm'
+                ],
+                extensionsLabels: ['.xls', '.xlsx']
+            },
+            {
+                mime: 'application/vnd.ms-powerpoint',
+                label: __('Microsoft Powerpoint'),
+                equivalent: [
+                    'application/vnd.ms-powerpoint.slideshow.macroEnabled.12',
+                    'application/vnd.openxmlformats-officedocument.presentationml.slideshow',
+                    'application/vnd.ms-powerpoint.presentation.macroEnabled.12',
+                    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+                    '.ppt',
+                    '.pptm',
+                    '.pptx',
+                    '.ppsm',
+                    '.ppsx'
+                ],
+                extensionsLabels: ['.ppt', '.pptx']
+            },
+            {
+                mime: 'application/vnd.oasis.opendocument.text',
+                label: __('OpenDocument text document'),
+                equivalent: ['.odt'],
+                extensionsLabels: ['.odt']
+            },
+            {
+                mime: 'application/vnd.oasis.opendocument.spreadsheet',
+                label: __('OpenDocument spreadsheet document'),
+                equivalent: ['.ods'],
+                extensionsLabels: ['.ods']
+            },
+            {
+                mime: 'text/x-c',
+                label: __('C++ file (.cpp)'),
+                equivalent: ['text/plain', '.cpp'],
```

Comment:
Should it be just `equivalent: ['.cpp']` or accepting any text file is intentional? 
Same for .cpp/.pas/.h

Reply:
I could go either way. Since a `.cpp` file is not guaranteed to be detected correctly on some systems, I made it more lenient. (Though if we check the extension as well, this may be pointless.)

Reply:
I myself like that we are being lenient, but before we discussed that we want validation to be strict. Strict would mean not accepting `.txt` with `text/plain`, but only accepting `.cpp` or `text/x-c`.

Reply:
Your point makes sense. I removed `text/plain` from these types.